messagebird/client.py

```diff
@@ -1,187 +1,255 @@
+        query = self._format_query(limit, offset)
+        return ContactList().load(self.request('contacts?' + query, 'GET', None))
+
+    def group(self, id):
+        return Group().load(self.request('groups/' + str(id), 'GET', None))
+
+    def group_create(self, name, params=None):
+        if params is None: params = {}
+        params.update({'name': name})
+        return Group().load(self.request('groups', 'POST', params))
+
+    def group_delete(self, id):
+        self.request_plain_text('groups/' + str(id), 'DELETE', None)
+
+    def group_list(self, limit=10, offset=0):
+        query = self._format_query(limit, offset)
+        return GroupList().load(self.request('groups?' + query, 'GET', None))
+
+    def group_update(self, id, name, params=None):
+        if params is None: params = {}
+        params.update({'name': name})
+        self.request_plain_text('groups/' + str(id), 'PATCH', params)
+
+    def group_add_contacts(self, groupId, contactIds):
+        query = self.__group_add_contacts_query(contactIds)
+        self.request_plain_text('groups/' + str(groupId) + '?' + query, 'PUT', None)
+
+    def __group_add_contacts_query(self, contactIds):
+        # __group_add_contacts_query gets a query string to add contacts to a
+        # group. The expected format is ids[]=first-contact&ids[]=second-contact.
+        # See: https://developers.messagebird.com/docs/groups#add-contact-to-group.
+        return '&'.join('ids[]=' + str(id) for id in contactIds)
+
+    def group_remove_contact(self, groupId, contactId):
+        self.request_plain_text('groups/' + str(groupId) + '/contacts/' + str(contactId), 'DELETE', None)
+
+    def conversation_list(self, limit=10, offset=0):
+        uri = CONVERSATION_PATH + '?' + self._format_query(limit, offset)
+        return ConversationList().load(self.request(uri, 'GET', None, CONVERSATION_TYPE))
+
+    def conversation_start(self, start_request):
+        uri = CONVERSATION_PATH + '/start'
+        return Conversation().load(self.request(uri, 'POST', start_request, CONVERSATION_TYPE))
+
+    def conversation_update(self, id, update_request):
+        uri = CONVERSATION_PATH + '/' + str(id)
+        return Conversation().load(self.request(uri, 'PATCH', update_request, CONVERSATION_TYPE))
+
+    def conversation_read(self, id):
+        uri = CONVERSATION_PATH + '/' + str(id)
+        return Conversation().load(self.request(uri, 'GET', None, CONVERSATION_TYPE))
+
+    def conversation_list_messages(self, conversation_id, limit=10, offset=0):
+        uri = CONVERSATION_PATH + '/' + str(conversation_id) + '/' + CONVERSATION_MESSAGES_PATH
+        uri += '?' + self._format_query(limit, offset)
+
+        return ConversationMessageList().load(self.request(uri, 'GET', None, CONVERSATION_TYPE))
+
+    def conversation_create_message(self, conversation_id, message_create_request):
+        uri = CONVERSATION_PATH + '/' + str(conversation_id) + '/' + CONVERSATION_MESSAGES_PATH
```

Comment:
Hm, I don't know if I like this way of string concatenation. '%s/%d/%s'.format(CONVERSATION_PATH, conversation_id, CONVERSATION_MESSAGES_PATH) - what do you think about this?

Reply:
I think in this case we should not update it, since all the client.py is using this concatenation with +. We can create a ticket to refactor all of them but I would prefer in a separate PR